src/io.ts

```diff
@@ -539,6 +542,34 @@ export class IO<E, A> {
         return IO.err(NonEmptyList.fromArray(errors));
       }
     });
+  }
+
+  /**
+   * Combines multiple `IO` operations into a single `IO` operation that, when executed, will run all
+   * operations in parallel and encapsulate their results in a tuple. If the operations succeed, the
+   * resulting `IO` instance will contain an `Ok` with a tuple of the results. If one or more operations
+   * fail, the resulting `IO` instance will contain an `Err` with a non-empty list of errors.
+   *
+   * @template E - The type of the error.
+   * @template T - A tuple type representing the types of the values contained in each IO instance.
+   * @param {...{[K in keyof T]: IO<E, T[K]>}} ops - A variable number of IO instances to be executed in parallel.
+   * @returns {IO<NonEmptyList<E>, T>} - A new IO instance that resolves to a tuple of successful results or a NonEmptyList of errors.
+   */
+  @Experimental()
+  static parZip<E, T extends any[]>(...ops: { [K in keyof T]: IO<E, T[K]> }): IO<NonEmptyList<E>, T> {
```

Comment:
Very nice, I wasn't aware of this language feature.
Just an idea: what if instead returning an tuple, which in my opinion is not that friendly to deal with, the functions takes another callback function with the tuple as the parameter lists, and returns an `IO<Nel<E>>, R>`, where the other function is something like `f: T -> R`?